### DISCUSSION TITLE
qhull Vertice Input Error

### DISCUSSION THREAD

**Discussion by Savillian** [mtex-toolbox/mtex#1221] [Ask Anything]
Hi everyone,

I was attempting to process an extremely large EBSD dataset (24 million datapoints) and ran into an error involving qhull verticies and calculating grain boundaries. I suspect the error (seen below) was due to the calculation process reaching some kind of limit, but I wasn't sure as to which limit this was and if a workaround existed? The code in question is observed below, but I suspect the issue doesn't lie in this.

Code:
`ebsd = EBSD.load(fname,CS,'interface','ang',...
  'convertEuler2SpatialReferenceFrame','setting 2'); % BD down screen

%% Clean Data
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd,'angle',2*degree);`

_**I wanted to ask if there is an upper limit to any given attribute for calculating boundaries in an EBSD map? I didn't run into any RAM issues like I have with larger datasets. Instead, qhull vertices seem to be the limiting factor as seen below.**_ 

**The Error Message**
_Error using matlab.internal.math.qhull
QH7074 qhull warning: more than 16777215 ridges.  ID field overflows and two ridges
may have the same identifier.  Otherwise output ok.
QH7074 qhull warning: more than 16777215 ridges.  ID field overflows and two ridges
may have the same identifier.  Otherwise output ok.
QH7074 qhull warning: more than 16777215 ridges.  ID field overflows and two ridges
may have the same identifier.  Otherwise output ok.
QH7074 qhull warning: more than 16777215 ridges.  ID field overflows and two ridges
may have the same identifier.  Otherwise output ok.
QH6159 qhull input error: more than 16777215 vertices.  ID field overflows and two vertices
may have the same identifier.  Vertices not sorted correctly.

While executing:  | qhull v Q5 Q6 Qs
Options selected for Qhull 2010.1 2010/01/14:
  run-id 681563457  voronoi  Q5-no-check-outer  Q6-no-concave-merge
  Qsearch-initial-simplex  _pre-merge  _zero-centrum  Pgood  _max-width 1.2e+07
  Error-roundoff 1.1e-08  _one-merge 7.7e-08  Visible-distance 2.2e-08
  U-coplanar-distance 2.2e-08  Width-outside 4.4e-08  _wide-facet 1.3e-07
Last point added to hull was p13683520.  Last merge was #23496378.

At error exit:

Voronoi diagram by the convex hull of 22456880 points in 3-d:

  Number of Voronoi regions: 16777214
  Total number of nearly incident points: 5679666
  Number of Voronoi vertices: 0
  Number of non-simplicial Voronoi vertices: 13271602

Statistics for:  | qhull v Q5 Q6 Qs

  Number of points processed: 16777215
  Number of hyperplanes created: 60419906
  Number of facets in hull: 19594723
  Number of distance tests for qhull: 1637348747
  Number of distance tests for merging: 96316094
  Number of distance tests for checking: 0
  Number of merged facets: 23496378

Error in voronoin (line 107)
[v, c] = matlab.internal.math.qhull(x', 'v ', opt);

Error in spatialDecomposition (line 35)
    [V,D] = voronoin([X;dummyCoordinates],{'Q5','Q6','Qs'}); %,'QbB'

Error in EBSD/calcGrains (line 62)
[V,F,I_FD] = spatialDecomposition([ebsd.prop.x(:), ebsd.prop.y(:)],ebsd.unitCell,varargin{:});

Error in BenReconstructionsandSchmidFactorsV9 (line 39)
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd,'angle',2*degree);_

**Comment by nyyssont** [comment]
Apparently there is a limit data when calculating convex hull. I found this:

https://githubmemory.com/repo/mtex-toolbox/mtex/issues/976

Where it's suggested to use the option 'noQHull' with calcGrains. Could you try that?
